### ISSUE TITLE
Regression: Incorrect error when using PEP 695 syntax

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#7716]
PEP 695 (and the typing spec) indicate:

> A compiler error or runtime exception is generated if the definition of an earlier type parameter references a later type parameter even if the name is defined in an outer scope.

```python
# The following generates no compiler error, but a type checker
# should generate an error because an upper bound type must be concrete,
# and ``Sequence[S]`` is generic. Future extensions to the type system may
# eliminate this limitation.
class ClassA[S, T: Sequence[S]]: ...

# The following generates no compiler error, because the bound for ``S``
# is lazily evaluated. However, type checkers should generate an error.
class ClassB[S: Sequence[T], T]: ...
```

The following test case in the conformance test "generics_syntax_scoping.py" now fails due to a recent regression.

```python
class ClassA[S, T: Sequence[S]]:  # E
    ...


class ClassB[S: Sequence[T], T]:  # E
    ...

S: int = int(0)
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.360, which I just published.
